includes/external-connection-cpt.php

```diff
@@ -541,7 +541,26 @@ function meta_box_external_connection_details( $post ) {
 				<tbody>
 					<?php foreach ( $post_types as $post_type ) : ?>
 						<?php
-						if ( 'dt_subscription' === $post_type->name ) {
+						/**
+						 * Filter to update the list of post types that should be hidden from the "Post types permissions" list.
+						 *
+						 * @since x.x.x
+						 * @hook dt_hide_post_types_from_permission_list
+						 *
+						 * @param {array} The list of post types..
+						 *
+						 * @return {bool} The updated array with the list of post types that should be hidden.
+						 */
+						$hide_from_list = apply_filters( 'dt_hide_post_types_from_permission_list', array() );
+
+						// Strictly hide 'dt_subscription' post type.
+						$hide_from_list[] = 'dt_subscription';
+
+						// Don't show:
+						// 1. Post types returning from a filter and 'dt_subscription'.
+						// 2. All _builtin post types except 'post' and 'page'.
+						if ( in_array( $post_type->name, $hide_from_list, true )
+							 || ( $post_type->_builtin && 'post' !== $post_type->name && 'page' !== $post_type->name ) ) {
```

Comment:
This has the effect of strictly hiding all built in post types. I can see a use case in which a site owner may wish to include resuable blocks or attachments in the list of types.

Instead, how about setting the default list prior to the filter:
```php
$hide_from_list = get_post_types( ['_builtin' => true, 'show_in_rest' => true] );
unset( $hide_from_list['post'], $hide_from_list['page'] );
$hide_from_list = array_values( $hide_from_list ); // Default is keyed by the post type 'post' => 'post', etc
```

This will make `dt_hide_post_types_from_permission_list` as flexible as possible.

Reply:
> This will make `dt_hide_post_types_from_permission_list` as flexible as possible.

I agree. Also, I think we should create a utility for this task. Putting all login inside HTML code (which is inside some `foreach` loops) makes it a bit hard to read IMO.

Reply:
> I agree. Also, I think we should create a utility for this task. Putting all login inside HTML code (which is inside some foreach loops) makes it a bit hard to read IMO.

@dinhtungdu excuse me, can you please elaborate more? are you suggesting moving the code to a separate file or something?

Reply:
@dinhtungdu that's interesting! Only the deference is we are using `show_in_rest => true`,

https://github.com/10up/distributor/blob/dcf23f33a0cd80bcf3be30f74066b291e80dbf62/includes/external-connection-cpt.php#L452-L457

And the utility function `distributable_post_types()` uses `public => true`.

https://github.com/10up/distributor/blob/dcf23f33a0cd80bcf3be30f74066b291e80dbf62/includes/utils.php#L283

Any reason why you have used `show_in_rest => true` (https://github.com/10up/distributor/commit/13614204987df5f85459a37236404cca0b8e6c26)? if you think we can use `public => true` instead, then IMO, it's fine to use the utility function. In that case, we also no need to introduce a new filter with this PR.